### ISSUE TITLE
Expose more stuff (Blob, File & FormData)

### ISSUE THREAD

**Issue by jimmywarting** [node-fetch/node-fetch#1454]
In other help section i have mention ways of posting a FormData, Blob, and a File

Some solutions involves adding same extra dependency on the same thing that node-fetch also depends on and other solutions involves more hacky stuff to get the internals of what node-fetch is using internally.

like: 
```js
import fetch, { Headers, Request, Response } from 'node-fetch'

// Get the internals fetch depends on
const { constructor: Blob } = await new Response().blob()
const { constructor: FormData } = await new Response(new URLSearchParams).formData()
const fd = new FormData(); fd.set('a', new Blob())
const { constructor: File } = fd.get('a')

const file = new File(['hello world'], 'sample.txt', { type: 'text/plain' })
fd.set('a', file)
fetch(url, { method: 'POST', body: fd })
```

it feels very hacky and unnecessary to go around the problem of using the same dependency as node-fetch is using. 
It also requires top level await to work...

I think that we should expose this 3 classes as well so you can just do:

```js
import fetch, { Headers, Request, Response, FormData, Blob, File } from 'node-fetch'

const file = new File(['hello world'], 'sample.txt', { type: 'text/plain' })
const fd = new FormData()
fd.set('a', file)
fetch(url, { method: 'POST', body: fd })
```
it would also be worth exposing the `fetch-blob/from.js` utility functions that turns a file into a Blob or a File without actually reading the hole content into memory... 

i think this would simplify the developers experiences and also making it easier to keep things up to date and avoiding dedupe and using different versions

---

For the v4 milestone, would it also be worth exposing the whatwg Readable/Wrtiable Stream as well?
(So that we take care of exposing nodes builtin whatwg stream whenever possible and falling back to web-streams-polyfill otherwise)

```js
import { stream } from 'node-fetch'
const { TransformStream, ReadableStream, WritableStream, ... } = stream
```
maybe it's too much exposure?
it is also possible to do the same hack here (but you only get ReadableStream)
```js
const ReadableStream = new Response('').body.constructor
const rs = new ReadableStream({ /* ... */ })
fetch(url, { method: 'POST', body: rs })
```
it would be nice to have the same instances of streams so they can be piped to...
piping from a native ReadableStream to a polyfilled WritableStream don't work well together...

**Comment by jimmywarting** [comment]
ping @LinusU @bitinn @gr2m @Richienb what do you think about this?

**Comment by gr2m** [comment]
I don't see any problem, I like it 👍🏼

**Comment by jimmywarting** [comment]
Thinking of doing this for v3 before we start working on v4 if anyone else is onboard with this suggestion

Maybe will just do: 
```js
// src/index.js

// exports { File, Blob, blobFrom, blobFromSync, fileFrom, fileFromSync }
export * from 'fetch-blob/from.js'
// exports FormData
export { FormData } from 'formdata-polyfill/esm.min.js'
```

and some minor readme changes

**Comment by LinusU** [comment]
Sounds good to me 👍

**Comment by Richienb** [comment]
Note: In a few months will be Node.js 12 EOL so the built-in [Blob](https://nodejs.org/api/buffer.html#class-blob) class in Node.js 14 could be used instead.

**Comment by jimmywarting** [comment]
- NodeJS Blob still have some failing WPT test when i run it last night
- NodeJS still lacks support for getting blobs backed up by the FileSystem
- NodeJS Blobs streaming method isn't so stream friendly cuz it reads all of the data into memory https://github.com/nodejs/node/blob/cea76dbf339bc6254ee170047267a76d0570f666/lib/internal/blob.js#L318-L326
they were to haisty to get something out and working just to follow the spec,
- NodeJS blobs also lowercase the type and loses the importans of meta key value like in:<br>`new Blob([], {type: 'multipart/form-data; boundary=----WebKitFormBoundaryr0nJrbD3GYWj6oSM'})` this boundary value is important to keep as is
- I'm unsure of how NodeJS blob binding to c++ works internally, the hole idea of a blob is that it's should have a strategy for offload memory to the disk. if you construct many large blobs that is never cleared, then it's meant to peg that memory to some temporary file if it's held for a longer time. 
- the only thing i think NodeJS Blob is useful for and what it got right is transferable between workers and creating ObjectURLS with `URL.createObjectURL` other that that i think fetch-blob is somewhat better cuz of better streaming and having blobs backed up by filesystem

**Comment by benmccann** [comment]
For SvelteKit, we'd like to standardize on using web streams for streaming if possible. Some of the serverless platforms we target such as Cloudflare have already done so ([docs](https://developers.cloudflare.com/workers/learning/using-streams)). Getting access to a web stream from `node-fetch` would really help in this regard and might be the main blocker that we face right now. In the short-term we could potentially polyfill it, though we might drop Node 14 support in the coming months, so might not need a polyfill longer-term.

**Comment by jimmywarting** [comment]
changing the `response.body` to a whatwg stream is a major breaking change... we like to change it eventually and it's planed for v4, at the same time v12 goes EOL soon. Until then i would like to try to fix as many issues as possible before then. 
we could start experimenting with it already such that we can accept a whatwg stream as a body.
